File: JavaAngular-profe/05/app/service/buscador.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Resultado } from '../model/Resultado';

@Injectable({
  providedIn: 'root'
})
export class BuscadorService {

  constructor(private http:HttpClient) { }

  buscarResultados(tematica:string):Observable<Resultado[]>{
    let url:string="http://localhost:7000/buscador/buscar";
    //llamada asíncrona
    return this.http.get<Resultado[]>(url+"?tematica="+tematica);
  }
}

File: JavaEE-servicio-rest/_01_servicio_buscador/src/main/java/service/interfaces/BuscadorService.java
package service.interfaces;

import java.util.List;

import model.Resultado;

public interface BuscadorService {
	List<Resultado> buscar(String tematica);

	void guardar(Resultado resultado);

	List<Resultado> eliminarResultado(String url);

	Resultado actualizarDescripcion(String url, String nuevaDescripcion);
}
